sql/core/src/test/scala/org/apache/spark/sql/execution/adaptive/AdaptiveQueryExecSuite.scala

```diff
@@ -1929,6 +1929,29 @@ class AdaptiveQueryExecSuite
       }
     }
   }
+
+  test("SPARK-36020: Check logical link in remove redundant projects") {
```

Comment:
For some reasons, this test is flaky or fails with Scala 2.13.
I filed a JIRA at https://issues.apache.org/jira/browse/SPARK-39525